cupy/manipulation/kind.py

```diff
@@ -19,6 +21,27 @@ def asfortranarray(a, dtype=None):
 
     """
     ret = cupy.empty(a.shape[::-1], a.dtype if dtype is None else dtype).T
+    if (a.flags.c_contiguous and
+            (a.dtype == numpy.float32 or a.dtype == numpy.float64) and
+            a.ndim == 2 and
+            dtype is None):
+        m, n = a.shape
+        if a.dtype == numpy.float32:
+            cupy.cuda.cublas.sgeam(
+                cupy.cuda.Device().cublas_handle,
+                1,  # transa
+                1,  # transb
+                m, n, 1., a.data.ptr, n, 0., a.data.ptr, n,
+                ret.data.ptr, m)
+        elif a.dtype == numpy.float64:
+            cupy.cuda.cublas.dgeam(
+                cupy.cuda.Device().cublas_handle,
+                1,  # transa
+                1,  # transb
+                m, n, 1., a.data.ptr, n, 0., a.data.ptr, n,
+                ret.data.ptr, m)
+        return ret
```

Comment:
I like
```python
else:
    ret[...] = a
    return ret
```
because it is easier to understand the above function considers two cases.